crypto/types/compact_bit_array_test.go

```diff
@@ -239,3 +241,35 @@ func BenchmarkNumTrueBitsBefore(b *testing.B) {
 		}
 	})
 }
+
+func TestNewCompactBitArrayCrashWithLimits(t *testing.T) {
+	if testing.Short() {
+		t.Skip("This test can be expensive in memory")
+	}
+	tests := []struct {
+		in       int
+		mustPass bool
+	}{
+		{int(^uint(0) >> 30), false},
+		{int(^uint(0) >> 1), false},
+		{int(^uint(0) >> 2), false},
+		{int(math.MaxInt32), true},
+		{int(math.MaxInt32) + 1, true},
+		{int(math.MaxInt32) + 2, true},
+		{int(math.MaxInt32) - 7, true},
+		{int(math.MaxInt32) + 24, true},
+		{int(math.MaxInt32) * 9, false}, // results in >=maxint after (bits+7)/8
+		{1, true},
+		{0, false},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(fmt.Sprintf("%d", tt.in), func(t *testing.T) {
```

Comment:
Add `t.Parallel()` inside `t.Run`, otherwise, the `tt := tt` is not necessary.

Reply:
It’s a precaution given that a bunch of this code changes yet we also don’t want parallel right now given that we allocate a bunch of memory. However for Go, it’d be the right suggestion :-) Thanks Cuong!